components/datetime/src/time_zone.rs

```diff
@@ -717,10 +809,23 @@ pub(super) enum IsoSeconds {
 }
 
 /// Whether a field should be zero-padded in ISO-8601 format.
-pub(super) enum ZeroPadding {
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum ZeroPadding {
     /// Add zero-padding.
     On,
 
     /// Do not add zero-padding.
     Off,
 }
+
+/// A config enum for initializing TimeZoneFormat.
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
```

Comment:
@sffc 

Are we okay with adding `Serialize` and `Deserialize` derives here? 

The only reason they exist is for testing. Do you know if these will be removed (essentially skipped) by dead code elimination?

Or should we add equivalent types only in the test module and implement `From<TestType> for ThisType`?

Reply:
@samchen61661 

Regardless of whether this can be ignored by DCE, it's failing our CI tests for building with/without all features enabled, because `serde` is not always available. 

So I believe we will have to do what I suggested above:

We need to remove the `Serialize` and `Deserialize` dependencies from `TimeZoneFormatConfig`, `IsoFormat`, `IsoMinutes`, `IsoSeconds`, and we need to make essentially the same types where we define the other (de)serializable sttructs for the test fixtures:

https://github.com/unicode-org/icu4x/blob/main/components/datetime/tests/patterns/structs/time_zones.rs

Then we need to implement the [`From`](https://doc.rust-lang.org/std/convert/trait.From.html) trait from the (de)serializable test types to turn into the real types in this file same.

So in `tests/patterns/structs/time_zones.rs` you would have something like this:

```rust
use icu_datetime::time_zone;

...

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub enum IsoSeconds {
    Optional,
    Never,
}

impl From<IsoSeconds> for time_zone::IsoSeconds {
    fn from(other: IsoSeconds) -> time_zone::IsoSeconds {
        match other {
            IsoSeconds::Optional => time_zone::IsoSeconds::Optional,
            IsoSeconds::Never => time_zone::IsoSeconds::Never,
        }
    }
}
```

Then do this for all of the enums, including `TimeZoneFormatConfig`

Implementing the `From` trait enables the `.into()` functionality as well, so when it comes time to do the `Iso8601` variant itself, you should be able to just do something like:

```rust
match other {
    ...
    Iso8601(iso_format, iso_minutes, iso_seconds) => time_zone::TimeZoneFormatConfig::Iso8601(
        iso_format.into(),
        iso_minutes.into(),
        iso_seconds.into(),
    ),
}
```